arangod/Aql/Functions.cpp

```diff
@@ -6944,8 +6944,24 @@ AqlValue Functions::DateFormat(arangodb::aql::Query* query,
   if (!aqlFormatString.isString()) {
     ::registerInvalidArgumentWarning(query, AFN);
     return AqlValue(AqlValueHintNull());
-  }      
+  }
 
   std::string const formatString = aqlFormatString.slice().copyString();
   return AqlValue(::executeDateFormatRegex(formatString, tp));
 }
+
+AqlValue Functions::Near(arangodb::aql::Query* query, transaction::Methods*,
+              VPackFunctionParameters const& params){
+    ::registerError(query,"NEAR",TRI_ERROR_QUERY_FUNCTION_ARGUMENT_TYPE_MISMATCH);
```

Comment:
This error type is wrong.
It should rather be something like TRI_ERROR_INTERNAL or TRI_ERROR_NOT_IMPLEMENTED with some descriptive message.
Same below for Within() and Fulltext()